File: ConsoleApp1/BusRoute.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MTS
{
    public class BusRoute
    {
        public int Id;
        public int _number;
        public string _name;
        private readonly List<int> _busStops = new List<int>();

        public BusRoute(int id, int number, string name)
        {
            Id = id;
            _name = name;
            _number = number;
        }

        public void ExtendRoute(int stopid)
        {
            _busStops.Add(stopid);
        }

        public Tuple<int, int> GetBusStopInfoByBusLocation(int location)
        {
            if (location >= _busStops.Count) return Tuple.Create(0, _busStops[0]);
            
            return Tuple.Create(location, _busStops[location]);
        }
    }
}

File: JavaPrototype/src/MTS/BusRoute.java
package MTS;

import javafx.util.Pair;

import java.util.ArrayList;

public class BusRoute {
    public Integer Id;
    public Integer _number;
    public String _name;
    private ArrayList _busStops = new ArrayList();

    public BusRoute(Integer id, Integer number, String name)
    {
        Id = id;
        _name = name;
        _number = number;
    }

    public void ExtendRoute(Integer stopid)
    {
        _busStops.add(stopid);
    }

    public Integer GetBusStopInfoByBusLocation(Integer location)
    {
        if (location >= _busStops.size()) return  (Integer) _busStops.get(0);

        return (Integer) _busStops.get(location);
    }

    public Integer GetBusLocationByBusLocation(Integer location)
    {
        if (location >= _busStops.size()) return 0;

        return location;
    }
}
